sdk/core/core-http/samples/gotHttpClient.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @summary This sample demonstrates how to implement a custom HttpClient.
+ */
+
+import { IncomingHttpHeaders } from "http";
+import * as coreHttp from "@azure/core-http";
+import got from "got";
+import { Headers as GotHeaders } from "got";
+
+/**
+ * Translate from core-http headers to got headers
+ * @param headers
+ * @returns
+ */
+function toOutgoingHeaders(headers: coreHttp.HttpHeadersLike): GotHeaders {
+  let result: Record<string, string> = {};
+  for (const key of headers.headerNames()) {
+    if (key) {
+      result[key] = headers.get(key) || "";
+    }
+  }
+
+  return result;
+}
+
+/**
+ * Translates got incoming headers into core-http headers
+ * @param headers
+ * @returns
+ */
+function toIncomingHeaders(headers: IncomingHttpHeaders): coreHttp.HttpHeadersLike {
+  const result = new coreHttp.HttpHeaders();
+  for (const key of Object.keys(headers)) {
+    if (Array.isArray(headers[key]) && headers[key]?.length) {
+      result.set(key, headers[key]![0]);
+    } else {
+      result.set(key, (headers[key] as string) || "");
+    }
+  }
```

Comment:
nit: Can make this a little cleaner

```suggestion
  for (const [key, value] of Object.entries(headers)) {
    if (Array.isArray(value) && value?.length) {
      result.set(key, value![0]);
    } else if (value) {
      result.set(key, value);
    }
  }
```